js/src/dag/tree.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const { series, eachSeries } = require('async')
```

Comment:
We use require('async/series') as a good practice to avoid requiring the whole module into our codebase.